Patch status: fixed in >=chromium-120.0.6088.0 / Qt6.8.0

Backport from [1] for [2] to fix a race condition.

[1] https://crrev.com/ac2ad9c9f01bf255fd123785bd37d3b0ec44e410
[2] https://bugs.gentoo.org/942142
--- a/src/3rdparty/chromium/third_party/blink/public/BUILD.gn
+++ b/src/3rdparty/chromium/third_party/blink/public/BUILD.gn
@@ -437,4 +437,5 @@
     "//third_party/blink/public/common",
     "//third_party/blink/public/common:headers",
+    "//third_party/blink/renderer/platform:make_platform_generated",
     "//ui/accessibility:ax_base",
     "//ui/base/dragdrop/mojom:mojom_shared",
--- a/src/3rdparty/chromium/third_party/blink/renderer/platform/BUILD.gn
+++ b/src/3rdparty/chromium/third_party/blink/renderer/platform/BUILD.gn
@@ -142,5 +142,8 @@
 group("make_platform_generated") {
   visibility = []  # Allow re-assignment of list.
-  visibility = [ "//third_party/blink/renderer/*" ]
+  visibility = [
+    "//third_party/blink/public:blink_headers",
+    "//third_party/blink/renderer/*",
+  ]
   public_deps = [
     ":bindings_buildflags",
@@ -151,4 +154,5 @@
     ":runtime_feature_state_override_context",
     "//third_party/blink/public/common:buildflags",
+    "//third_party/blink/renderer/platform/heap:blink_heap_buildflags",
     "//third_party/blink/renderer/platform/network:make_generated",
   ]
@@ -197,5 +201,4 @@
     "//third_party/blink/public/mojom:mojom_platform_blink",
     "//third_party/blink/public/mojom:web_bluetooth_mojo_bindings_blink",
-    "//third_party/blink/renderer/platform/heap:blink_heap_buildflags",
     "//third_party/blink/renderer/platform/network:make_generated",
     "//third_party/blink/renderer/platform/wtf",
